src/framework/backend/winit.rs

```diff
@@ -111,6 +111,12 @@ where
         slog_scope::logger(),
     );
 
+    ddata
+        .get::<Anodium>()
+        .unwrap()
+        .config
+        .output_new(output.clone());
+
```

Comment:
Oh, here Anodium is kept in ddata intentionally to keep it separate from the backend, the basic idea is for the backend to not know what Anodium is, and be more or less self-contained module.
I also overlooked that in previous udev PR: https://github.com/PolyMeilex/Anodium/blob/d7f1f017197e51deefc4699f9fe70e996d4e24db/src/framework/backend/udev.rs#L402
But in udev it's a little bit complex, so let's ignore it for now.

But here it's super easy to avoid, just move `output_new` call to the event handler side
- for winit specific stuff here inside of winit callback: https://github.com/PolyMeilex/Anodium/blob/d7f1f017197e51deefc4699f9fe70e996d4e24db/src/framework/backend/mod.rs#L59
- if it's something every backend needs, here: https://github.com/PolyMeilex/Anodium/blob/06bfba0d07f94247124272325f51b70e99bd6571/src/backend_handler.rs#L8

Reply:
I think you mean that, right https://github.com/dragonnn/Anodium/blob/calllop-async/src/framework/backend/winit.rs#L114 ?
About the udev, yeah that isn't that easy, to avoid switching the resolution after something is already displayed we need to return somehow back the mode the BackendEvent resolves. I assume they are not executed right away when you call cb?

Reply:
Or maybe introduce a new BackenEvent that will request the compositor to select mode, that would work for both udev and winit, because udev has to know the mode before it creates the initial surface output.
So the events would be sent it this order:
- BackendEvent::RequestOutputConfigure 
- BackendEvent::OutputCreated 
Or something along those lines...

Reply:
The last commit should fix that for winit.
But now in udev on_new will be called twice.
As far I understand if `(inner.cb)(BackendEvent::OutputCreated { output }, ddata.reborrow());` would be moved inside scan_connectors on the line 402 it would work like it should, since as far I see what inner.cb does it runs the function right away, I am righ?

Reply:
If RequestOutputConfigure will give a regular Output structure it should work perfectly fine

Reply:
But then OutputCreated shouldn't call output_new in config

Reply:
it's a plain callback so it runs right away, so it should be fine to call it whenever, and the order will be kept

Reply:
Ok, now it should work fine and no backend uses Anodium state :)